qe/drivers.py

```diff
@@ -1707,3 +1866,682 @@ def selection_step(
 
     # 3.
     return (*Powerplant(lewis, psi_det_extented).E_and_psi_coef, psi_det_extented)
+
+
+#  ___  _________ _____
+#  |  \/  || ___ \_   _|
+#  | .  . || |_/ / | |
+#  | |\/| ||  __/  | |
+#  | |  | || |    _| |_
+#  \_|  |_/\_|    \___/
+#
+
+
+def dispatch_psi(comm, psi_i):
+    """Function to partition and dispatch pieces of trial wavefunction psi_i to worker processes
+
+    :param comm: mpi4py.MPI.COMM_WORLD communicator object
+    :param psi_i: Psi_det, list of determinants"""
+
+    rank = comm.Get_rank()  # Rank of current process
+    size = comm.Get_size()  # No. of processes currently running
+
+    if rank == 0:
+        indices = np.arange(len(psi_i), dtype="i")
+        number_of_dets, res = divmod(len(psi_i), size)
+        # Number of determinants sent to each worker; First `res` processes sent one extra
+        sendcounts = np.array(
+            [number_of_dets + 1 if i < res else number_of_dets for i in range(size)]
```

Comment:
please put the `dtype` in all the the np.array. This avoid nasty bugs :) (and will made the porting to C++ easier)